lms/djangoapps/teams/views.py

```diff
@@ -79,16 +81,22 @@ def has_team_api_access(user, course_key):
     Args:
       user (User): The user to check access for.
       course_key (CourseKey): The key to the course which we are checking access to.
+      access_username (string): If provided, access_username must match user.username for non staff access.
 
     Returns:
       bool: True if the user has access, False otherwise.
     """
-    return (CourseEnrollment.is_enrolled(user, course_key) or
-            CourseStaffRole(course_key).has_user(user) or
-            user.is_staff)
+    if user.is_staff:
+        return True
+    if CourseStaffRole(course_key).has_user(user):
+        return True
+    if ((access_username == user.username if access_username else True) and
```

Comment:
I find this logic hard to read. I'm thinking you don't need the ``if`` because ``user.username`` will never be falsey. How about:

if access_username == user.username:
    return CourseEnrollment.is_enrolled(user, course_key)